js/lib/appUrlUtil.js

```diff
@@ -124,6 +124,44 @@ module.exports.isTargetAboutUrl = function (input) {
 }
 
 /**
+ * Gets a local URL that renders the given magnet:// URL
+ * Returns null if the input is not a magnet URL
+ * Example: getTargetMagnetUrl('magnet:?x=y') -> 'chrome-extension://<...>.html#magnet:?x=y'
+ */
+module.exports.getTargetMagnetUrl = function (input) {
+  if (!input.startsWith('magnet:?')) return null
+  const url = module.exports.getAppUrl('webtorrent.html')
+  return url + '#' + input
+}
+
+/**
+ * Gets the magnet:// URL for a given local URL.
+ * Returns null if the input is not the local URL for a magnet link.
+ * Example: getSourceMagnetUrl('chrome-extension://<...>.html#magnet:?x=y') -> 'magnet:?x=y'
+ */
+module.exports.getSourceMagnetUrl = function (input) {
+  const url = module.exports.getAppUrl('webtorrent.html')
+  if (!input.startsWith(url)) return null
+  return input.substring(input.indexOf('#') + 1)
```

Comment:
The target URL is internal-only and controlled by us, so I think that we can assume it will always be a magnet link or torrent file. This seems to be the assumption that is made by the analogous `getSourceAboutUrl` function. Correct me if I'm wrong?